### PR TITLE
Jbl/l4t rebased

### PR DESCRIPTION
**Pull request by jennerl**
## Status
**READY/IN DEVELOPMENT/HOLD**

## Migrations
YES | NO

## Description
This PR adds support for NVIDIA's L4T-based hardware. The purpose of this work is to exercise Remek's deqtify branch and also to prove our platform-sdk codebase is capable of running on a popular embedded system.

## Related PRs
This PR is based on the deqtify branch (already merged to master) so is based on that branch. There is some more work toward making sure the base branch works with QGC that will be related, but is not yet authored, and does not depend on this PR.

branch | PR
------ | ------
other_pr_develop | [link]()
other_pr_master | [link]()

## Todos
- [ ] Tests
- [ ] Documentation

## Notes
General notes regarding the pull request

## Impacted Areas in AirMap Platform SDK
List general components of the SDK that this PR will affect:
* no platform-sdk code was changed, though I did modernise boringssl by 2.5 years and updated cpp-jwt submodule to a modern version

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -8,3 +8,4 @@
 [submodule "vendor/cpp-jwt"]
 	path = vendor/cpp-jwt
 	url = https://github.com/arun11299/cpp-jwt.git
+	branch = 8272256dd0105e2e66838dc081e30e638966ca12
```
Comment (bluecamel) [inline .gitmodules:11]: They have this tagged as `v1.4`, so you could change to that for clarity.

File: vendor/boringssl/CMakeLists.txt [hunk 2]
```diff
@@ -16,7 +16,7 @@ endif ()
 
 ExternalProject_add(
     boringssl
-    URL file://${CMAKE_CURRENT_SOURCE_DIR}/boringssl-fips-20180730.tar.gz
+    GIT_REPOSITORY https://github.com/google/boringssl.git
```
Comment (bluecamel) [inline vendor/boringssl/CMakeLists.txt:19]: Might want to add a `GIT_TAG` with a commit hash (doesn't look like they tag releases).

File: docker/l4t [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+RUN tools/l4t/setup.dev.sh
+RUN mkdir build
+WORKDIR build
+RUN cmake -DCMAKE_INSTALL_PREFIX=/usr -DAIRMAP_ENABLE_NETWORK_TESTS=OFF --trace-source=CMakeLists.txt --trace-expand ..
+RUN make -j $(nproc)
+RUN ctest -V
+RUN make install
+RUN make package
+RUN mv *.deb /tmp
+WORKDIR /usr/src/app
```
Comment (bluecamel) [inline docker/l4t:18]: It seems that everything from this line forward is not needed.

File: tools/l4t/build-docker-image.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -eux
+HERE="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+uname -r
+sudo apt-get install qemu binfmt-support qemu-user-static
+
+build_opts="--force-rm=true $@"
+docker run --platform=linux/amd64 --rm --privileged multiarch/qemu-user-static --reset -p yes
```
Comment (bluecamel) [inline tools/l4t/build-docker-image.sh:10]: Why is qemu needed here?  I got the impression that this was handled by the L4T base image.

File: .gitmodules (hunk 1 above)
Comment (jennerl) [reply .gitmodules]: updated to reference tag instead

File: docker/l4t (hunk 3 above)
Comment (jennerl) [reply docker/l4t:18]: the cleanup script and `rm -rf` command clean up the image so it's not so big..not sure how big a deal this is, but will save some space on your FS.

i'll remove the WORKDIR and ENTRYPOINT commands

File: tools/l4t/build-docker-image.sh (hunk 4 above)
Comment (jennerl) [reply tools/l4t/build-docker-image.sh:10]: my understanding is that the container is based on aarm64 architecture and qemu is necessary to emulate aarm64 on x86_64.

I based some of my work on [what stereolabs did](https://www.stereolabs.com/docs/docker/building-arm-container-on-x86/)

File: vendor/boringssl/CMakeLists.txt (hunk 2 above)
Comment (jennerl) [reply vendor/boringssl/CMakeLists.txt:19]: good idea...changing now...done

**Review by bluecamel — APPROVED** [review]

**Pull request merged**